File: topic_4/Python/12-splitints.py
# Recebe uma lista e retorna uma tupla com uma lista de ímpares
# e outra lista de pares

def splitints(xs):
    listaimpar = []
    listapar = []
    tupla = ()
    for x in xs:
        if x % 2 != 0:
            listaimpar.append(x)
        if x % 2 == 0:
            listapar.append(x)
    tupla = (listaimpar, listapar)
    return tupla

print(splitints ([1,2,3,4,5,6,7]) == ([1,3,5,7],[2,4,6])) #True
print(splitints ([2,4,6,1,1,7]) == ([1,1,7],[2,4,6])) #True

File: topic_4/Haskell/12-splitints.hs
-- Recebe uma lista e retorna uma tupla com uma lista de ímpares
-- e outra lista de pares

-- pega so os numeros impares
splitimpar xs = [x | x <- xs, impar x]
    where
        impar x = x `mod` 2 /= 0

-- pega so os numeros pares
splitpar xs = [x | x <- xs, par x]
    where
        par x = x `mod` 2 == 0

-- tamanho da lista de impares
tamimpar xs = length(splitimpar xs)

-- Retorna a tupla das listas de impares e pares juntas
splitints xs = splitAt (tamimpar xs) (juntarlistas (splitimpar xs) (splitpar xs))
    where
        juntarlistas xs ys = xs ++ ys

main = do
    a <- readLn :: IO [Int]
    print $ splitints a